test/built-ins/Atomics/wait/not-an-object-throws.js

```diff
@@ -0,0 +1,41 @@
+// Copyright (C) 2018 Amal Hussein. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+/*---
+esid: sec-map-objects
+description: >
+  the typedArray param must be an object TODO
+info: |
+  Atomics.wait( typedArray, index, value, timeout )
+
+  1.Let buffer be ? ValidateSharedIntegerTypedArray(typedArray, true).
+    ...
+    2. if Type(typedArray) is not Object, throw a TypeError exception
+features: [ TODO ]
+---*/
+
+//TODO: note this test is valid in an environments when [[CanBlock]] is true
+
+var nonObjectTypes = [
+  null,
+  undefined,
+  true,
+  false,
+  "**string**",
+  Number.NEGATIVE_INFINITY,
+  -14,
+  0,
+  14,
+  3.14,
+  Number.POSITIVE_INFINITY,
+  Symbol('**symbol**')
+];
+
+
+for(let nonObject of nonObjectTypes) {
+  try {
+    assert.throws(TypeError, () => Atomics.wait(nonObject,0,0,0)); //TODO used function here instead of => ?
```

Comment:
please, expand this in individual assertions, it's even easier to read...

```js
assert.throws(TypeError, function() { Atomics.wait(null, 0, 0, 0); }, 'null');
assert.throws(TypeError, function() { Atomics.wait(undefined, 0, 0, 0); }, 'undefined');
assert.throws(TypeError, function() { Atomics.wait(true, 0, 0, 0); }, 'true');
assert.throws(TypeError, function() { Atomics.wait(false, 0, 0, 0); }, 'false');
```

No need to expand it to 6 different numbers, one should suffice and maybe add `NaN` as well.